src/TypeScript.UnitTests/Rules/RulesProviderFactoryTests.cs

```diff
@@ -138,6 +143,23 @@ public void Create_GetDefinitions_WithDefaultParams_ReturnsExpectedConfigs()
             result[3].DefaultParams.Should().HaveCount(2);
         }
 
+        [TestMethod]
```

Comment:
Minor: I'm not sure this test belongs here. It's part of the internal RuleProvider implementation, not the factory.

Reply:
Yes and no.  The detailed behaviour of the RuleProvider is tested elsewhere.
However, we have no way of testing that the factory is producing valid artefacts except to test those artefacts i.e. it's more like an integration test.

I could perhaps have added some assertions relating to the mocked users settings to check that the created RuleProvider was initialized as expected.